ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1050 @@
+
+    fn try_from(value: FramebufferFormat) -> Result<Self, Self::Error> {
+        match value {
+            FramebufferFormat::Rgb565 => Ok(CamOutputFormat::RGB_565),
+            _ => Err(())
+        }
+    }
+}
+
+impl TryFrom<CamOutputFormat> for FramebufferFormat {
+    type Error = ();
+
+    fn try_from(value: CamOutputFormat) -> Result<Self, Self::Error> {
+        match value {
+            CamOutputFormat::RGB_565 => Ok(FramebufferFormat::Rgb565),
+            _ => Err(())
+        }
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamShutterSoundType: u32 {
+        const NORMAL    = ctru_sys::SHUTTER_SOUND_TYPE_NORMAL;
+        const MOVIE     = ctru_sys::SHUTTER_SOUND_TYPE_MOVIE;
+        const MOVIE_END = ctru_sys::SHUTTER_SOUND_TYPE_MOVIE_END;
+    }
+}
+
+pub struct CamTrimmingParams {
+    pub x_start: i16,
+    pub y_start: i16,
+    pub x_end: i16,
+    pub y_end: i16,
+}
+
+#[derive(Default)]
+pub struct ImageQualityCalibrationData(ctru_sys::CAMU_ImageQualityCalibrationData);
+
+#[derive(Default)]
+pub struct StereoCameraCalibrationData(ctru_sys::CAMU_StereoCameraCalibrationData);
+
+#[derive(Default)]
+pub struct PackageParameterCameraSelect(ctru_sys::CAMU_PackageParameterCameraSelect);
+
+#[derive(Default)]
+pub struct PackageParameterContext(ctru_sys::CAMU_PackageParameterContext);
+
+#[derive(Default)]
+pub struct PackageParameterContextDetail(ctru_sys::CAMU_PackageParameterContextDetail);
+
+impl Cam {
+    pub fn init() -> crate::Result<Cam> {
+        unsafe {
+            let r = ctru_sys::camInit();
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(Cam(()))
+            }
+        }
+    }
+
+    pub fn start_capture(&self, port: CamPort) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_StartCapture(port.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn stop_capture(&self, port: CamPort) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_StopCapture(port.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn is_busy(&self, port: CamPort) -> crate::Result<bool> {
+        unsafe {
+            let mut is_busy = false;
+            let r = ctru_sys::CAMU_IsBusy(&mut is_busy, port.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(is_busy)
+            }
+        }
+    }
+
+    pub fn clear_buffer(&self, port: CamPort) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_ClearBuffer(port.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn get_vsync_interrupt_event(&self, port: CamPort) -> crate::Result<u32> {
```

Comment:
Maybe it would be clearer to return the `Handle` type?
(same comment for the other similar functions)

Reply:
With the talk about the `SVC` module, it seems it may be best to not expose handles like this for safety. If `SVC` functions stay internal, the user will never have a need for getting handles and therefore any handle related function should probably not be exposed or even removed if never used.

Reply:
I don't think it's a concern here to just use the type from `ctru_sys`. It doesn't change anything relevant to safety. But, this code might change anyway based on @Meziu's comments.